### PR TITLE
Keyboard Navigation on Modal View & Keyboard Navigation for Tags

### PR DESCRIPTION
**Pull request by guptaharsh25**
[AB#34321](https://microsoftgarage.visualstudio.com/002806e2-ebaa-4672-9d2e-5fe5d29154ef/_workitems/edit/34321)
[AB#34356](https://microsoftgarage.visualstudio.com/002806e2-ebaa-4672-9d2e-5fe5d29154ef/_workitems/edit/34356)

This PR ensures that when a modal is open, buttons in the background are not keyboard accessible.

Also addressed making tagging feature keyboard accessible.

### CODE REVIEW

File: private-templates-service/client/src/components/Common/Tags/Tags.tsx [hunk 1]
```diff
@@ -74,7 +77,7 @@ class Tags extends React.Component<Props, State>  {
         this.highlightTag(tag, this.props.tags);
       }
       else if (tag === "") {
-        this.closeAddTag();
+        //this.closeAddTag();
```
Comment (beervoley) [inline private-templates-service/client/src/components/Common/Tags/Tags.tsx]: Should this be here?

Comment (guptaharsh25) [reply private-templates-service/client/src/components/Common/Tags/Tags.tsx]: i'll remove it, thanks

File: private-templates-service/client/src/components/Common/Tags/Tags.tsx [hunk 2]
```diff
@@ -118,20 +133,20 @@ class Tags extends React.Component<Props, State>  {
     const {
       isAdding
     } = this.state;
-
+    console.log(this.props.modalState);
```
Comment (beervoley) [inline private-templates-service/client/src/components/Common/Tags/Tags.tsx]: please remove this:)

File: private-templates-service/client/src/components/SideBar/SideBar.tsx [hunk 3]
```diff
@@ -181,7 +184,7 @@ const SideBar = (props: Props) => {
         {props.isAuthenticated && <NavMenu styles={navMenuLinksProps} groups={navMenuLinks} onLinkClick={onNavClick} />}
       </MainItems>
 
-      <SignOut onClick={props.authButtonMethod}>Sign {props.isAuthenticated ? "Out" : "In"}</SignOut>
+      <SignOut onClick={props.authButtonMethod} tabIndex={props.modalState ? -1 : 0}>Sign {props.isAuthenticated ? "Out" : "In"}</SignOut>
```
Comment (beervoley) [inline private-templates-service/client/src/components/SideBar/SideBar.tsx]: What is this for?

File: private-templates-service/client/src/components/Common/Tags/Tags.tsx (hunk 2 above)
Comment (guptaharsh25) [reply private-templates-service/client/src/components/Common/Tags/Tags.tsx]: removed

File: private-templates-service/client/src/components/SideBar/SideBar.tsx (hunk 3 above)
Comment (guptaharsh25) [reply private-templates-service/client/src/components/SideBar/SideBar.tsx]: when a modal is open, users are unable to sign out by clicking on the sign out button, but if you use keyboard navigation, you can sign out by tabbing onto the sign out button.

this simply checks if there is a modal currently open and then if it is you can't tab onto the sign out button.

i did the same thing for all the buttons on template preview page

File: private-templates-service/client/src/components/Common/Tags/Tags.tsx [hunk 4]
```diff
@@ -73,9 +76,7 @@ class Tags extends React.Component<Props, State>  {
       if (this.props.tags.includes(tag)) {
         this.highlightTag(tag, this.props.tags);
       }
-      else if (tag === "") {
-        this.closeAddTag();
-      }
+      else if (tag === "") { }
```
Comment (beervoley) [inline private-templates-service/client/src/components/Common/Tags/Tags.tsx]: If condition with an empty body - do we need it or there's another way to organize the logic?

File: private-templates-service/client/src/components/Dashboard/PreviewModal/PreviewModal.tsx [hunk 5]
```diff
@@ -127,7 +130,7 @@ class PreviewModal extends React.Component<Props, State> {
         {template && !isFetching ?
           <React.Fragment>
             <ACOuterPanel>
-              <StyledDropdown selectedKey={selectedItem.key} onChange={this.hostConfigChange} options={DropdownOptions} />
+              <StyledDropdown selectedKey={selectedItem.key} onChange={this.hostConfigChange} options={DropdownOptions} tabIndex={this.props.modalState ? -1 : 0} />
```
Comment (beervoley) [inline private-templates-service/client/src/components/Dashboard/PreviewModal/PreviewModal.tsx]: Can I ask - why do we need tabIndex here? This dropdown only appears when the preview is open and there's no other case when it's visible

File: private-templates-service/client/src/components/Common/Tags/Tags.tsx (hunk 4 above)
Comment (guptaharsh25) [reply private-templates-service/client/src/components/Common/Tags/Tags.tsx]: done

**Pull request merged**